### DISCUSSION TITLE
How I can load and predict With Flask?

### DISCUSSION THREAD

**Discussion by DevanshMistry890** [yangheng95/PyABSA#252] [Q&A]
Hello, I am wanting to make a front end for the ABSA project, so how can I dump/save the model so later I can load it while runtime in flask? I tried to save checkpoints but it is not working. Any help will be appreciated.

**Comment by yangheng95** [comment]
Generally, the checkpoints will be saved if you set checkpoint_save_mode in a trainer interface. What exception did you get while saving a checkpoint?

**Reply by DevanshMistry890** [reply]
from google.protobuf import descriptor as _descriptor
ImportError: cannot import name 'descriptor'
----------------------------------
I keep getting this error whenever I load local checkpoints, I have created a new environment and installed the package before running but still shows an error.

**Reply by yangheng95** [reply]
Did run the code on Colab?

**Reply by DevanshMistry890** [reply]
I am wanting to work with a flask, so tested it locally.

**Reply by DevanshMistry890** [reply]
Protobuf also has a native os version so, after installing it above error is solved. Now checkpoints are working. But how can I load my model in the flask, I tried to call using torch.load but not working.

**Reply by yangheng95** [reply]
Could you please paste your code snippet and the console oupout which is needed to locate prolem?

**Reply by DevanshMistry890** [reply]
how I can serve this model using flask rest API?
